host_reaper.py

```diff
@@ -76,6 +76,23 @@ def run(config, logger, session, event_producer, notification_event_producer, sh
     with application.app.app_context():
         filter_hosts_to_delete = find_hosts_in_state(logger, session, ["culled"])
 
+        # Adhoc fix for RHINENG-16901
+        # hosts reporter by rhsm-system-profile-bridge are not being deleted
+        # when marked as culled, and are also prevented to stay culled as w
+        # we are forcing its modified_on and last_check_in to be updated.
+
+        rhsm_briedge_hosts = (
+            session.query(Host)
+            .filter(and_(or_(False, *filter_hosts_to_delete), Host.reporter == "rhsm-system-profile-bridge"))
+            .all()
+        )
+        filter_hosts_to_delete = filter_hosts_to_delete + [Host.reporter != "rhsm-system-profile-bridge"]
+        rhsm_briedge_hosts_ids = []
+        for host in rhsm_briedge_hosts:
+            host._update_modified_date()
+            host._update_last_check_in_date()
+            rhsm_briedge_hosts_ids.append(host.id)
```

Comment:
```suggestion
            rhsm_bridge_hosts_ids.append(host.id)
```
Also, what's the purpose of creating this huge list of IDs? It doesn't appear to be used anywhere